### PR TITLE
add Table Name to genserver state

### PR DESCRIPTION
**Pull request by avelooleg**

### CODE REVIEW

**Comment by Lol4t0** [comment]
Fix tests please.
Travis build should pass

**Comment by avelooleg** [comment]
All checks have passed

**Comment by avelooleg** [comment]
All checks have passed

File: src/indexed_cache.erl [hunk 1]
```diff
@@ -22,7 +22,8 @@
 -type objects() :: list(object()).
 -type object() :: term().
 
--type poolid() ::atom().
+-type poolid() :: atom().
```
Comment (Lol4t0) [inline src/indexed_cache.erl:25]: atom

File: src/indexed_cache_connection.erl [hunk 2]
```diff
@@ -11,18 +11,18 @@
...
--record(state, {pool_id, field_types, field_names}).
+-record(state, {pool_id, table_name, field_types, field_names}).
 
-start_link(PoolId, Fieldtypes, FieldNames, ConnectionOpts) ->
-    gen_server:start_link({local, PoolId}, ?MODULE, [PoolId, Fieldtypes, FieldNames, ConnectionOpts], []).
+start_link(PoolId, TableName, Fieldtypes, FieldNames, ConnectionOpts) ->
+    gen_server:start_link({local, PoolId}, ?MODULE, [PoolId, TableName, Fieldtypes, FieldNames, ConnectionOpts], []).
 
-init([PoolId, FieldTypes, FieldNames, ConnectionOpts]) when is_tuple(FieldTypes), is_tuple(FieldNames) ->
+init([PoolId, TableName, FieldTypes, FieldNames, ConnectionOpts]) when is_binary(TableName), is_tuple(FieldTypes), is_tuple(FieldNames) ->
```
Comment (Lol4t0) [inline src/indexed_cache_connection.erl:22]: or binary?

File: test/indexed_cache_request_tst.erl [hunk 3]
```diff
@@ -83,7 +83,8 @@ invalid_request_test_() ->
 setup() ->
     meck:new(indexed_cache_connection),
     meck:expect(indexed_cache_connection, field_names, 1, indexed_cache_data:field_names(data)),
-    meck:expect(indexed_cache_connection, field_types, 1, indexed_cache_data:field_types(data)).
+    meck:expect(indexed_cache_connection, field_types, 1, indexed_cache_data:field_types(data)),
+	meck:expect(indexed_cache_connection, table_name, 1, table_name).
```
Comment (Lol4t0) [inline test/indexed_cache_request_tst.erl]: or atom?

**Comment by Lol4t0** [comment]
Also please reindent everything with spaces with default tab size=4.

Tabs indentation is environment-dependent, therefore may look ugly sometimes (on github, for example :-)

File: src/indexed_cache.erl (hunk 1 above)
Comment (avelooleg) [inline src/indexed_cache.erl:25]: poolid is atom, table_name is binary

File: src/indexed_cache_connection.erl (hunk 2 above)
Comment (avelooleg) [inline src/indexed_cache_connection.erl:22]: poolid is atom, table_name is binary

File: test/indexed_cache_request_tst.erl (hunk 3 above)
Comment (avelooleg) [inline test/indexed_cache_request_tst.erl]: Here you can use whatever. Function just should not be undefined to go to the next call on the list. I'll change it  for binary for sanity sake.

**Pull request merged**